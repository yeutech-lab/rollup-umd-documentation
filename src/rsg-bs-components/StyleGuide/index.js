export { default } from 'rsg-components/StyleGuide/StyleGuideRenderer';
