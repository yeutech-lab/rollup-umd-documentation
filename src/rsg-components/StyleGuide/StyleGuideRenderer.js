import React from 'react';
import PropTypes from 'prop-types';
import omit from 'lodash.omit';
import bp from 'bootstrap-styled-mixins/lib/breakpoints';
import styled from 'styled-components';
import mapToCssModules from 'map-to-css-modules/lib';
import cn from 'classnames';
import SideBar from '../../components/SideBar';
import FooterRenderer from '../../components/FooterRenderer';
import Ribbon from '../Ribbon';

export const defaultProps = {
  theme: {
    styleguide: {
      '$rsg-styleguide-background-color': '#fff',
      '$rsg-styleguide-has-sidebar-padding-left': {
        xs: '0',
        md: '250px',
      },
      '$rsg-styleguide-content-max-width': '1000px',
      '$rsg-styleguide-content-padding': {
        xs: '16px',
        md: '16px 95px 16px 32px',
      },
      '$rsg-styleguide-content-margin': '0 auto',
      '$rsg-styleguide-content-display': 'block',
      '$rsg-styleguide-sidebar-background-color': '#f5f5f5',
      '$rsg-styleguide-sidebar-border': '#e8e8e8 solid',
      '$rsg-styleguide-sidebar-border-width': {
        xs: '1px 0 0 0',
        md: '0 1px 0 0',
      },
      '$rsg-styleguide-sidebar-position': {
        xs: 'static',
        md: 'fixed',
      },
      '$rsg-styleguide-sidebar-top': '0',
      '$rsg-styleguide-sidebar-left': '0',
      '$rsg-styleguide-sidebar-bottom': '0',
      '$rsg-styleguide-sidebar-width': {
        xs: 'auto',
        md: '250px',
      },
      '$rsg-styleguide-sidebar-overflow': 'auto',
      '$rsg-styleguide-sidebar-padding-bottom': '4px',
      '$rsg-styleguide-logo-padding': '16px',
      '$rsg-styleguide-logo-border-bottom': '1 #e8e8e8 solid',
      '$rsg-styleguide-footer-display': 'block',
      '$rsg-styleguide-footer-color': '#767676',
      '$rsg-styleguide-footer-font-family': '-apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif',
      '$rsg-styleguide-footer-font-size': '13px',
    },
  },
};

/* eslint-disable react/require-default-props */
export const propTypes = {
  /**
   * @ignore
   */
  className: PropTypes.string, // eslint-disable-line react/require-default-props
  title: PropTypes.string.isRequired,
  homepageUrl: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  toc: PropTypes.node.isRequired,
  hasSidebar: PropTypes.bool, // eslint-disable-line react/require-default-props
  logo: PropTypes.node,
  logohref: PropTypes.string,
  /** Theme variables. Can be: */
  theme: PropTypes.shape({
    styleguide: PropTypes.shape({
      '$rsg-styleguide-background-color': PropTypes.string,
      '$rsg-styleguide-has-sidebar-padding-left': PropTypes.shape({
        xs: PropTypes.string,
        md: PropTypes.string,
      }),
      '$rsg-styleguide-content-max-width': PropTypes.string,
      '$rsg-styleguide-content-padding': PropTypes.shape({
        xs: PropTypes.string,
        md: PropTypes.string,
      }),
      '$rsg-styleguide-content-margin': PropTypes.string,
      '$rsg-styleguide-content-display': PropTypes.string,
      '$rsg-styleguide-sidebar-background-color': PropTypes.string,
      '$rsg-styleguide-sidebar-border': PropTypes.string,
      '$rsg-styleguide-sidebar-border-width': PropTypes.shape({
        xs: PropTypes.string,
        md: PropTypes.string,
      }),
      '$rsg-styleguide-sidebar-position': PropTypes.shape({
        xs: PropTypes.string,
        md: PropTypes.string,
      }),
      '$rsg-styleguide-sidebar-top': PropTypes.string,
      '$rsg-styleguide-sidebar-left': PropTypes.string,
      '$rsg-styleguide-sidebar-bottom': PropTypes.string,
      '$rsg-styleguide-sidebar-width': PropTypes.shape({
        xs: PropTypes.string,
        md: PropTypes.string,
      }),
      '$rsg-styleguide-sidebar-overflow': PropTypes.string,
      '$rsg-styleguide-sidebar-padding-bottom': PropTypes.string,
      '$rsg-styleguide-logo-padding': PropTypes.string,
      '$rsg-styleguide-logo-border-bottom': PropTypes.string,
      '$rsg-styleguide-footer-display': PropTypes.string,
      '$rsg-styleguide-footer-color': PropTypes.string,
      '$rsg-styleguide-footer-font-family': PropTypes.string,
      '$rsg-styleguide-footer-font-size': PropTypes.string,
    }),
  }),
  /**
   * Replace or remove a className from the component.
   * See example <a href="https://www.npmjs.com/package/map-to-css-modules" target="_blank">here</a>.
   */
  cssModule: PropTypes.object, // eslint-disable-line react/require-default-props
};
/* eslint-enable react/require-default-props */


const StyleGuideRendererUnstyled = (props) => {
  const {
    className,
    title,
    homepageUrl,
    children,
    toc,
    hasSidebar,
    logo,
    logohref,
    cssModule,
    ...attributes
  } = omit(props, ['theme']);

  return (
    <div
      className={mapToCssModules(cn(
        className,
        'rsg-styleguide',
        (hasSidebar && 'has-sidebar')
      ), cssModule)}
      {...attributes}
    >
      {hasSidebar && (
        <SideBar logo={{ logo, href: logohref }} title={title} items={toc} />
      )}
      <Ribbon />
      <main className="content">
        {children}
        {hasSidebar && (
          <FooterRenderer />
        )}
      </main>

    </div>
  );
};

StyleGuideRendererUnstyled.defaultProps = defaultProps;
StyleGuideRendererUnstyled.propTypes = propTypes;

const StyleGuideRenderer = styled(StyleGuideRendererUnstyled)` 
  ${(props) => `
    &.rsg-styleguide {
      background-color: ${props.theme.styleguide['$rsg-styleguide-background-color']};
      .content {
        max-width: ${props.theme.styleguide['$rsg-styleguide-content-max-width']};
        margin: ${props.theme.styleguide['$rsg-styleguide-content-margin']};
        display: ${props.theme.styleguide['$rsg-styleguide-content-display']};
  ${bp.up(
    'xs',
    props.theme['$grid-breakpoints'],
    `
      padding: ${props.theme.styleguide['$rsg-styleguide-content-padding'].xs};
    `
  )}
  ${bp.up(
    'md',
    props.theme['$grid-breakpoints'],
    `
      padding: ${props.theme.styleguide['$rsg-styleguide-content-padding'].md};
    `
  )}
        .footer {
          footer-display: ${props.theme.styleguide['$rsg-styleguide-footer-display']};
          color: ${props.theme.styleguide['$rsg-styleguide-footer-color']};
          font-family: ${props.theme.styleguide['$rsg-styleguide-footer-font-family']};
          font-size: ${props.theme.styleguide['$rsg-styleguide-footer-font-size']};
        }
      }
      .sidebar {
        background-color: ${props.theme.styleguide['$rsg-styleguide-sidebar-background-color']};
        border: ${props.theme.styleguide['$rsg-styleguide-sidebar-border']};
        border-width: ${props.theme.styleguide['$rsg-styleguide-sidebar-border-width']};
        top: ${props.theme.styleguide['$rsg-styleguide-sidebar-top']};
        left: ${props.theme.styleguide['$rsg-styleguide-sidebar-left']};
        bottom: ${props.theme.styleguide['$rsg-styleguide-sidebar-bottom']};
        overflow: ${props.theme.styleguide['$rsg-styleguide-sidebar-overflow']};
  ${bp.up(
    'xs',
    props.theme['$grid-breakpoints'],
    `
      position: ${props.theme.styleguide['$rsg-styleguide-sidebar-position'].xs};
      width: ${props.theme.styleguide['$rsg-styleguide-sidebar-width'].xs};
      border-width: ${props.theme.styleguide['$rsg-styleguide-sidebar-border-width'].xs};
      padding-bottom: ${props.theme.styleguide['$rsg-styleguide-sidebar-padding-bottom']};
    `
  )}
  ${bp.up(
    'md',
    props.theme['$grid-breakpoints'],
    `
      position: ${props.theme.styleguide['$rsg-styleguide-sidebar-position'].md};
      width: ${props.theme.styleguide['$rsg-styleguide-sidebar-width'].md};
      border-width: ${props.theme.styleguide['$rsg-styleguide-sidebar-border-width'].md};
    `
  )}
        .logo {
          padding: ${props.theme.styleguide['$rsg-styleguide-logo-padding']};
          border-bottom: ${props.theme.styleguide['$rsg-styleguide-logo-border-bottom']};
        }
      }
    }
    &.has-sidebar {
  ${bp.up(
    'xs',
    props.theme['$grid-breakpoints'],
    `
      padding-left: ${props.theme.styleguide['$rsg-styleguide-has-sidebar-padding-left'].xs};
    `
  )}
  ${bp.up(
    'md',
    props.theme['$grid-breakpoints'],
    `
      padding-left: ${props.theme.styleguide['$rsg-styleguide-has-sidebar-padding-left'].md};
    `
  )}
    }
 `}
`;

StyleGuideRenderer.defaultProps = defaultProps;
StyleGuideRenderer.propTypes = propTypes;

export default StyleGuideRenderer;
