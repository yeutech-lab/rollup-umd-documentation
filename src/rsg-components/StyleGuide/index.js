export { default } from '../StyleGuide/StyleGuide';
