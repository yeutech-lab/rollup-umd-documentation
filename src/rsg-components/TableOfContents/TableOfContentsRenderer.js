import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import cn from 'classnames';
import omit from 'lodash.omit';
import mapToCssModules from 'map-to-css-modules/lib';
import Nav from 'bootstrap-styled/lib/Nav';
import Form from 'bootstrap-styled/lib/Form';
import Input from 'bootstrap-styled/lib/Input';


export const defaultProps = {
  theme: {
    styleguide: {
      '$rsg-toc-display': 'block',
      '$rsg-toc-padding': '0',
      '$rsg-toc-form-width': '60%',
      '$rsg-toc-form-margin': '0 auto',
    },
  },
};

export const propTypes = {
  className: PropTypes.string, // eslint-disable-line react/require-default-props
  children: PropTypes.node, // eslint-disable-line react/require-default-props
  searchTerm: PropTypes.string.isRequired,
  onSearchTermChange: PropTypes.func.isRequired,
  /** Theme variables. Can be: */
  theme: PropTypes.shape({
    styleguide: PropTypes.shape({
      '$rsg-toc-display': PropTypes.string,
      '$rsg-toc-padding': PropTypes.string,
      '$rsg-toc-form-width': PropTypes.string,
      '$rsg-toc-form-margin': PropTypes.string,
    }),
  }),
  /**
   * Replace or remove a className from the component.
   * See example <a href="https://www.npmjs.com/package/map-to-css-modules" target="_blank">here</a>.
   */
  cssModule: PropTypes.object, // eslint-disable-line react/require-default-props
};

const TableOfContentsRendererUnstyled = (props) => {
  const {
    className,
    children,
    searchTerm,
    onSearchTermChange,
    cssModule,
    ...attributes
  } = omit(props, ['theme']);
  return (
    <div
      className={mapToCssModules(cn(className, 'rsg-toc'), cssModule)}
      {...attributes}
    >
      <Form className="rsg-toc-form">
        <Input
          className="rsg-toc-form-input"
          size="sm"
          value={searchTerm}
          placeholder="Filter by name"
          aria-label="Filter by name"
          onChange={(event) => onSearchTermChange(event.target.value)}
        />
      </Form>
      <Nav className="rsg-toc-nav">
        {children}
      </Nav>
    </div>
  );
};

TableOfContentsRendererUnstyled.defaultProps = defaultProps;
TableOfContentsRendererUnstyled.propTypes = propTypes;

const TableOfContentsRenderer = styled(TableOfContentsRendererUnstyled)`
  ${(props) => `
    &.rsg-toc {
      .rsg-toc-form {
        background: #B31255;
        padding: 0 0 20px 0;
        .rsg-toc-form-input {
          width: ${props.theme.styleguide['$rsg-toc-form-width']};
          margin: ${props.theme.styleguide['$rsg-toc-form-margin']};
        }
      }
      .rsg-toc-nav {
        padding: 20px 0 0 0 !important;
        background: white;
        text-align: left;
        display: ${props.theme.styleguide['$rsg-toc-display']};
        padding: ${props.theme.styleguide['$rsg-toc-padding']};
      }
    }
  `}
`;

TableOfContentsRenderer.defaultProps = defaultProps;
TableOfContentsRenderer.propTypes = propTypes;

export default TableOfContentsRenderer;
