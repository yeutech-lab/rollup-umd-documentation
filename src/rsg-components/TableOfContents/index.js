export { default } from '../TableOfContents/TableOfContents';
