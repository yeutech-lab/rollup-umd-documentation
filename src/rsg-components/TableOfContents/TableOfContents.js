import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ComponentsList from 'rsg-components/ComponentsList';
import TableOfContentsRenderer from 'rsg-components/TableOfContents/TableOfContentsRenderer';
import filterSectionsByName from '../../utils/filterSectionsByName';

export default class TableOfContents extends Component {
	static propTypes = {
		sections: PropTypes.array.isRequired,
		useIsolatedLinks: PropTypes.bool,
	};
	state = {
		searchTerm: '',
	};

	renderLevel(sections, useIsolatedLinks = false) {
		const items = sections.map(section => {
			const children = [...(section.sections || []), ...(section.components || [])];
			return Object.assign({}, section, {
				heading: !!section.name && children.length > 0,
				content: children.length > 0 && this.renderLevel(children, useIsolatedLinks),
			});
		});
		return <ComponentsList items={items} useIsolatedLinks={useIsolatedLinks} />;
	}

	renderSections() {
		const { searchTerm } = this.state;
		const { sections, useIsolatedLinks } = this.props;
		// If there is only one section, we treat it as a root section
		// In this case the name of the section won't be rendered and it won't get left padding
		const firstLevel = sections.length === 1 ? sections[0].components : sections;
		const filtered = filterSectionsByName(firstLevel, searchTerm);

		return this.renderLevel(filtered, useIsolatedLinks);
	}

	render() {
		const { searchTerm } = this.state;
		return (
			<TableOfContentsRenderer
				searchTerm={searchTerm}
				onSearchTermChange={searchTerm => this.setState({ searchTerm })}
			>
				{this.renderSections()}
			</TableOfContentsRenderer>
		);
	}
}
