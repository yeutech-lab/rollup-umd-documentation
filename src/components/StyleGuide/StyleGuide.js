import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TableOfContents from 'react-styleguidist/lib/rsg-components/TableOfContents';
import StyleGuideRenderer from 'react-styleguidist/lib/rsg-components/StyleGuide/StyleGuideRenderer';
import Sections from 'react-styleguidist/lib/rsg-components/Sections';
import Welcome from 'react-styleguidist/lib/rsg-components/Welcome';
import Error from 'react-styleguidist/lib/rsg-components/Error';
import { HOMEPAGE } from 'react-styleguidist/scripts/consts';
import { DisplayModes } from 'react-styleguidist/lib/consts';

/**
 * This function will return true, if the sidebar should be visible and false otherwise.
 *
 * These sorted conditions (highest precedence first) define the visibility
 * state of the sidebar.
 *
 * - Sidebar is hidden for isolated example views
 * - Sidebar is always visible when pagePerSection
 * - Sidebar is hidden when showSidebar is set to false
 * - Sidebar is visible when showSidebar is set to true for non-isolated views
 *
 * @param {boolean} displayMode
 * @param {boolean} showSidebar
 * @param {boolean} pagePerSection
 * @returns {boolean}
 */
function hasSidebar(displayMode, showSidebar, pagePerSection = false) {
	return (
		(pagePerSection && displayMode !== DisplayModes.example) ||
		(showSidebar && displayMode === DisplayModes.all)
	);
}

export default class StyleGuide extends Component {
	static propTypes = {
	  /** Define if component should rerender (1) or not (0) */
		codeRevision: PropTypes.number.isRequired,
    /** Config from styleguide.config.js file */
		config: PropTypes.object.isRequired,
    /** File from rsg-components/slots/index.js */
		slots: PropTypes.object.isRequired,
    /** Array of section to show defined in styleguide.ext.json */
		sections: PropTypes.array.isRequired,
		welcomeScreen: PropTypes.bool,
		patterns: PropTypes.array,
		displayMode: PropTypes.string,
		allSections: PropTypes.array.isRequired,
		pagePerSection: PropTypes.bool,
	};

	static childContextTypes = {
	  /* @codeRevision:  */
		codeRevision: PropTypes.number.isRequired,
		config: PropTypes.object.isRequired,
		slots: PropTypes.object.isRequired,
		displayMode: PropTypes.string,
	};

	static defaultProps = {
		displayMode: DisplayModes.all,
	};

	state = {
		error: false,
		info: null,
	};

	getChildContext() {
		return {
			codeRevision: this.props.codeRevision,
			config: this.props.config,
			slots: this.props.slots,
			displayMode: this.props.displayMode,
		};
	}

	componentDidCatch(error, info) {
		this.setState({
			error,
			info,
		});
	}

	render() {
		const {
			config,
			sections,
			welcomeScreen,
			patterns,
			displayMode,
			allSections,
			pagePerSection,
		} = this.props;

		if (this.state.error) {
			return <Error error={this.state.error} info={this.state.info} />;
		}

		if (welcomeScreen) {
			return <Welcome patterns={patterns} />;
		}

		return (
			<StyleGuideRenderer
        fee={console.log(config.title)}
				title={config.title}
				homepageUrl={HOMEPAGE}
				toc={<TableOfContents sections={allSections} useIsolatedLinks={pagePerSection} />}
				hasSidebar={hasSidebar(displayMode, config.showSidebar, pagePerSection)}
			>
				<Sections sections={sections} depth={1} />
			</StyleGuideRenderer>
		);
	}
}
