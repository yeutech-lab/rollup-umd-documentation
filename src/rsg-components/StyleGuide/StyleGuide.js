import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { HOMEPAGE } from 'react-styleguidist/scripts/consts';
import { DisplayModes } from 'react-styleguidist/lib/consts';
import StyleGuideRenderer from './StyleGuideRenderer';
import TableOfContents from '../TableOfContents';
import Sections from '../Sections';
import Welcome from '../Welcome';
import Error from '../Error';

/**
 * This function will return true, if the sidebar should be visible and false otherwise.
 *
 * These sorted conditions (highest precedence first) define the visibility
 * state of the sidebar.
 *
 * - Sidebar is hidden for isolated example views
 * - Sidebar is always visible when pagePerSection
 * - Sidebar is hidden when showSidebar is set to false
 * - Sidebar is visible when showSidebar is set to true for non-isolated views
 *
 * @param {boolean} displayMode
 * @param {boolean} showSidebar
 * @param {boolean} pagePerSection
 * @returns {boolean}
 */
function hasSidebar(displayMode, showSidebar, pagePerSection = false) {
  return (
    (pagePerSection && displayMode !== DisplayModes.example) ||
    (showSidebar && displayMode === DisplayModes.all)
  );
}

export default class StyleGuide extends Component {
  /* eslint-disable react/require-default-props */
  static propTypes = {
    codeRevision: PropTypes.number.isRequired,
    config: PropTypes.object.isRequired,
    slots: PropTypes.object.isRequired,
    sections: PropTypes.array.isRequired,
    welcomeScreen: PropTypes.bool,
    patterns: PropTypes.array,
    displayMode: PropTypes.string,
    allSections: PropTypes.array.isRequired,
    pagePerSection: PropTypes.bool,
  };
  /* eslint-enable react/require-default-props */
  static childContextTypes = {
    codeRevision: PropTypes.number.isRequired,
    config: PropTypes.object.isRequired,
    slots: PropTypes.object.isRequired,
    displayMode: PropTypes.string,
  };

  static defaultProps = {
    displayMode: DisplayModes.all,
  };

  state = {
    error: false,
    info: null,
  };

  getChildContext() {
    return {
      codeRevision: this.props.codeRevision,
      config: this.props.config,
      slots: this.props.slots,
      displayMode: this.props.displayMode,
    };
  }

  componentDidCatch(error, info) {
    this.setState({
      error,
      info,
    });
  }

  render() {
    const {
      config,
      sections,
      welcomeScreen,
      patterns,
      displayMode,
      allSections,
      pagePerSection,
    } = this.props;

    if (this.state.error) {
      return <Error error={this.state.error} info={this.state.info} />;
    }

    if (welcomeScreen) {
      return <Welcome patterns={patterns} />;
    }

    return (
      <StyleGuideRenderer
        title={config.title}
        homepageUrl={HOMEPAGE}
        toc={<TableOfContents sections={allSections} useIsolatedLinks={pagePerSection} />}
        hasSidebar={hasSidebar(displayMode, config.showSidebar, pagePerSection)}
      >
        <Sections sections={sections} depth={1} />
      </StyleGuideRenderer>
    );
  }
}
