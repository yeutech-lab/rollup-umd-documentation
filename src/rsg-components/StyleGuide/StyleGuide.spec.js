import React from 'react';
import TableOfContents from '../TableOfContents';
import StyleGuide from './StyleGuide';
import { StyleGuideRenderer } from './StyleGuideRenderer';
import { DisplayModes } from '../../consts';

const sections = [
	{
		components: [
			{
				name: 'Foo',
				pathLine: 'components/foo.js',
				filepath: 'components/foo.js',
				props: {
					description: 'Foo foo',
				},
			},
			{
				name: 'Bar',
				pathLine: 'components/bar.js',
				filepath: 'components/bar.js',
				props: {
					description: 'Bar bar',
				},
			},
		],
	},
];
const config = {
	title: 'Hello',
	showSidebar: true,
};

it('should render components list', () => {
	const actual = shallow(
		<StyleGuide
			codeRevision={1}
			config={config}
			sections={sections}
			allSections={sections}
			slots={{}}
		/>
	);

	expect(actual).toMatchSnapshot();
});

it('should render welcome screen', () => {
	const actual = shallow(
		<StyleGuide
			codeRevision={1}
			config={config}
			sections={[]}
			allSections={[]}
			slots={{}}
			welcomeScreen
		/>
	);

	expect(actual).toMatchSnapshot();
});

it('should render an error when componentDidCatch() is triggered', () => {
	const wrapper = shallow(
		<StyleGuide codeRevision={1} config={config} sections={[]} allSections={[]} slots={{}} />
	);
	wrapper
		.instance()
		.componentDidCatch({ toString: () => 'error' }, { componentStack: { toString: () => 'info' } });
	wrapper.update();
	expect(wrapper).toMatchSnapshot();
});

describe('sidebar rendering', () => {
	it('renderer should have sidebar if showSidebar is not set', () => {
		const wrapper = shallow(
			<StyleGuide
				codeRevision={1}
				config={config}
				sections={sections}
				allSections={sections}
				slots={{}}
			/>
		);

		expect(wrapper.prop('hasSidebar')).toEqual(true);
	});

	it('renderer should not have sidebar if showSidebar is false', () => {
		const wrapper = shallow(
			<StyleGuide
				codeRevision={1}
				config={{
					...config,
					showSidebar: false,
				}}
				sections={sections}
				allSections={sections}
				slots={{}}
			/>
		);

		expect(wrapper.prop('hasSidebar')).toEqual(false);
	});

	it('renderer should not have sidebar in isolation mode', () => {
		const wrapper = shallow(
			<StyleGuide
				codeRevision={1}
				config={config}
				sections={sections}
				allSections={sections}
				slots={{}}
				displayMode={DisplayModes.component}
			/>
		);

		expect(wrapper.prop('hasSidebar')).toEqual(false);
	});

	it('renderer should have sidebar if pagePerSection is true', () => {
		const wrapper = shallow(
			<StyleGuide
				codeRevision={1}
				config={config}
				sections={sections}
				allSections={sections}
				slots={{}}
				displayMode={DisplayModes.component}
				pagePerSection
			/>
		);

		expect(wrapper.prop('hasSidebar')).toEqual(true);
	});
});

it('renderer should render logo, table of contents, ribbon and passed children', () => {
	const actual = shallow(
		<StyleGuideRenderer
			classes={{}}
			title={config.title}
			toc={<TableOfContents sections={sections} />}
			homepageUrl="http://react-styleguidist.js.org/"
			hasSidebar
		>
			<h1>Content</h1>
		</StyleGuideRenderer>
	);

	expect(actual).toMatchSnapshot();
});
