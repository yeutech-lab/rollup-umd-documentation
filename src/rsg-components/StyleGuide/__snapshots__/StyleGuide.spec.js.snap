// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render logo, table of contents, ribbon and passed children 1`] = `
<div
  className=""
>
  <main>
    <h1>
      Content
    </h1>
    <footer>
      <Markdown
        text="Generated with [React Styleguidist](http://react-styleguidist.js.org/)"
      />
    </footer>
  </main>
  <div>
    <div>
      <Styled(Logo)>
        Hello
      </Styled(Logo)>
    </div>
    <TableOfContents
      sections={
        Array [
          Object {
            "components": Array [
              Object {
                "filepath": "components/foo.js",
                "name": "Foo",
                "pathLine": "components/foo.js",
                "props": Object {
                  "description": "Foo foo",
                },
              },
              Object {
                "filepath": "components/bar.js",
                "name": "Bar",
                "pathLine": "components/bar.js",
                "props": Object {
                  "description": "Bar bar",
                },
              },
            ],
          },
        ]
      }
    />
  </div>
  <Ribbon />
</div>
`;

exports[`should render an error when componentDidCatch() is triggered 1`] = `
<Styled(Error)
  error={
    Object {
      "toString": [Function],
    }
  }
  info={
    Object {
      "componentStack": Object {
        "toString": [Function],
      },
    }
  }
/>
`;

exports[`should render components list 1`] = `
<Styled(StyleGuide)
  hasSidebar={true}
  homepageUrl="https://react-styleguidist.js.org/"
  title="Hello"
  toc={
    <TableOfContents
      sections={
        Array [
          Object {
            "components": Array [
              Object {
                "filepath": "components/foo.js",
                "name": "Foo",
                "pathLine": "components/foo.js",
                "props": Object {
                  "description": "Foo foo",
                },
              },
              Object {
                "filepath": "components/bar.js",
                "name": "Bar",
                "pathLine": "components/bar.js",
                "props": Object {
                  "description": "Bar bar",
                },
              },
            ],
          },
        ]
      }
      useIsolatedLinks={undefined}
    />
  }
>
  <Sections
    depth={1}
    sections={
      Array [
        Object {
          "components": Array [
            Object {
              "filepath": "components/foo.js",
              "name": "Foo",
              "pathLine": "components/foo.js",
              "props": Object {
                "description": "Foo foo",
              },
            },
            Object {
              "filepath": "components/bar.js",
              "name": "Bar",
              "pathLine": "components/bar.js",
              "props": Object {
                "description": "Bar bar",
              },
            },
          ],
        },
      ]
    }
  />
</Styled(StyleGuide)>
`;

exports[`should render welcome screen 1`] = `<Styled(Welcome) />`;
